compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpResourceValidator.java

```diff
@@ -298,6 +299,9 @@ public static void extractInputParamTypeAndValidate(SyntaxNodeAnalysisContext ct
                         reportInvalidIntersectionObjectType(ctx, paramLocation, paramName, typeName);
                         continue;
                     }
+                    if (isEnumQueryParamType(typeSymbol)) {
```

Comment:
within the `isEnumQueryParamType`, this kind == TypeDescKind.TYPE_REFERENCE check is done again, which is not needed. Simply move this if to L307 and check if the typeDescKind equals enum

Reply:
`TypeDescKind` results in `Union` for `enums`. Only the `SymbolKind` can be used to identify an `enum`.